### PR TITLE
packages: dont queue auto-index job if no extsvc of given kind

### PR DESCRIPTION
**Pull request by Strum355**
Previously, there was an unhandled case where if no external service was returned when listing external services for the kind associated with a package repo auto-indexing scheduling job, we would attempt to queue auto-index jobs anyways. This would cause a lot of error logging for repos not existing in the table but being requested for.

This PR makes us early-out if no external service was found, and also includes an improvement to handling package repos that dont exist (`len(repoNames)` >= `len(results)`, may be useful down the line if/when we dont add repo entries for non-existant packages. Do we want this in a separate PR perhaps?). 

## Test plan

Ran locally, and regressions should hopefully be handled by unit tests

### CODE REVIEW

**Review by mrnugget — APPROVED** [review]

File: enterprise/internal/codeintel/autoindexing/internal/background/job_dependency_indexing_scheduler.go [hunk 1]
```diff
@@ -96,6 +96,12 @@ func (h *dependencyIndexingSchedulerHandler) Handle(ctx context.Context, logger
 			return errors.Wrap(err, "extsvcStore.List")
 		}
 
+		if len(externalServices) == 0 {
```
Comment (ryanslade) [inline enterprise/internal/codeintel/autoindexing/internal/background/job_dependency_indexing_scheduler.go:99]: How much work would it be to add a test for this?

**Review by ryanslade — APPROVED** [review]

File: enterprise/internal/codeintel/autoindexing/internal/background/job_dependency_indexing_scheduler.go (hunk 1 above)
Comment (Strum355) [reply enterprise/internal/codeintel/autoindexing/internal/background/job_dependency_indexing_scheduler.go:99]: Very little, ill do that right after lunch

Comment (Strum355) [reply enterprise/internal/codeintel/autoindexing/internal/background/job_dependency_indexing_scheduler.go:99]: Unit test added

**Review by sashaostrikov — COMMENTED** [review]
Nice, thanks for fixing it!

**Pull request merged**